### DISCUSSION TITLE
How to change immediate text display layer? Seems stuck at 64

### DISCUSSION THREAD

**Discussion by hsandt** [DmitriySalnikov/godot_debug_draw_3d#54] [Q&A] [answered]
Hello, thanks for making this plugin! I just installed it today and I tried to draw a simple label, but it appeared below my HUD.

My HUD CanvasLayer was on layer 100, so I reduced the number to compare. I found out that it is still over the immediate GUI at 64, but goes under it at 63.

I searched source code for those numbers, to no avail. I don't see a setting to change it, and the priority parameter doesn't affect it (I think priority is more for ordering text).

Is there a way I can tune the display layer of the text?

Fortunately 64 is high enough, I can just put my HUD in the middle e.g. 32. But it's still be good to know in case I work on another project using high layer numbers.

**Comment by hsandt** [comment]
Ah, I could confirm my hypothesis by inspecting the remote tree, where DebugDrawManager is on CanvasLayer 64:

![image](https://github.com/user-attachments/assets/d43037ff-17ac-4661-ad30-002220713b2e)

![image](https://github.com/user-attachments/assets/f6d5f144-d4c3-4962-b80d-caba62e6c1a0)

However, it's not an autoload singleton so I don't have direct access, but it shouldn't be too complicated to find that node and set CanvasLayer layer if needed.

**Comment by DmitriySalnikov** [comment, marked as answer]
You can use [DebugDraw2D.custom_canvas](https://dd3d.dmitriysalnikov.ru/docs/1.4.4/classDebugDraw2D.html#a0b111f429b856cb93df850211e620238) on the desired layer.

An example from the demo:
```gdscript
DebugDraw2D.custom_canvas = %CustomCanvas if Input.is_key_pressed(KEY_ALT) else null
```

https://github.com/user-attachments/assets/68bcf950-d2de-4239-9c6f-cffc0c66596e

---

I feel dumb, but I only thought about it after the previous part of the comment..

Just change `layer` of `DebugDrawManager`:

https://github.com/user-attachments/assets/f4a62f5d-a64b-462d-820e-60ab19cf377f
